apps/glusterfs/dbcommon.go

```diff
@@ -227,3 +229,23 @@ func DeleteBricksWithEmptyPath(db *bolt.DB, all bool, clusterIDs []string, nodeI
 	}
 	return nil
 }
+
+// OpenDB is a wrapper over bolt.Open. It takes a bool to decide whether it should be a read-only open.
+// Other bolt DB config options remain local to this function.
+func OpenDB(dbfilename string, ReadOnly bool) (dbhandle *bolt.DB, err error) {
```

Comment:
Small nitpick: All other things being equal I'd prefer to see two helper functions like OpenDB / OpenDBReadOnly rather than using an argument. In this case it would be good because the argument is never selected dynamically anywhere in the code.